system-addon/lib/TopSitesFeed.jsm

```diff
@@ -72,13 +72,24 @@ this.TopSitesFeed = class TopSitesFeed {
         this.getScreenshot(link.url);
       }
     }
-
+    return links;
+  }
+  async refresh(action) {
+    let links = await this.getLinks();
     const newAction = {type: at.TOP_SITES_UPDATED, data: links};
 
     // Send an update to content so the preloaded tab can get the updated content
     this.store.dispatch(ac.SendToContent(newAction, action.meta.fromTarget));
```

Comment:
If the concern of checking if `fromTarget` exists, the method could probably just take an optional target instead of an `action`. If there's a target, `SendToContent` and if not, it's a `BroadcastToContent`